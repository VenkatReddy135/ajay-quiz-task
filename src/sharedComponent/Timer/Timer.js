import React, { useState, useEffect } from "react";
import Button from "../Button/Button";

const Timer = (props) => {
  const [time, setTime] = useState(200000);

  useEffect(() => {
    setTimeout(() => {
      setTime(time - 1000);
    }, 1000);
  }, [time]);

  const timeLeft = {
    minutes: Math.floor(time / 60 / 1000) % 60,
    seconds: Math.floor(time / 1000) % 60,
  };

  if (timeLeft.minutes === 0 && timeLeft.seconds === 0) {
    props.viewResults();
  }

  return (
    <div
      className="container"
      style={{ marginTop: "2em", marginBottom: "1em" }}
    >
      <Button
        buttonClass="btn btn-info"
        buttonValue={`${timeLeft.minutes}:${timeLeft.seconds}`}
      />
    </div>
  );
};
export default Timer;
