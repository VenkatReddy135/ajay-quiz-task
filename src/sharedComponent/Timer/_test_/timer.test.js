/* eslint-disable testing-library/no-node-access */
import React from "react";
import { cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Timer from "../Timer";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

describe("Timer component", () => {
  const renderedData = renderer.create(<Timer />).toJSON();

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Timer />, div);
  });

  it("renders a snapshot", () => {
    expect(renderedData).toMatchSnapshot();
  });

  afterEach(() => {
    cleanup();
  });
});
