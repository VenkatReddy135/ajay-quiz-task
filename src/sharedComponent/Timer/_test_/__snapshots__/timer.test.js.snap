// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer component renders a snapshot 1`] = `
<div
  className="container"
  style={
    Object {
      "marginBottom": "1em",
      "marginTop": "2em",
    }
  }
>
  <button
    className="btn btn-info"
    type="button"
    value=""
  >
    3:20
  </button>
</div>
`;
